src/Shared/HttpSys/RequestProcessing/PathNormalizer.cs

```diff
@@ -13,7 +15,7 @@ internal static class PathNormalizer
         // In-place implementation of the algorithm from https://tools.ietf.org/html/rfc3986#section-5.2.4
         public static unsafe int RemoveDotSegments(Span<byte> input)
         {
-            fixed (byte* start = input)
+            fixed (byte* start = &MemoryMarshal.GetReference(input))
```

Comment:
Spoke with @anurse and this is unnecessary in 2.2 since the compiler does this optimization automatically. Will revert.